TEST: The test test_io.py fails on HDK

### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)


### Reproducible Example

```python
MODIN_STORAGE_FORMAT=hdk python -m pytest modin/pandas/test/test_io.py::TestExcel::test_read_excel_header_none
```


### Issue Description

The test fails on HDK.

### Expected Behavior

Do not fail.

### Error Logs

<details>

```python-traceback

    def test_read_excel_header_none(self):
>       eval_io(
            fn_name="read_excel",
            io="modin/pandas/test/data/every_other_row_nan.xlsx",
            header=None,
        )

modin/pandas/test/test_io.py:1896: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
modin/experimental/core/execution/native/implementations/hdk_on_native/test/utils.py:63: in eval_io
    general_eval_io(
modin/pandas/test/utils.py:898: in eval_io
    call_eval_general()
modin/pandas/test/utils.py:880: in call_eval_general
    eval_general(
modin/pandas/test/utils.py:830: in eval_general
    comparator(*values)
modin/experimental/core/execution/native/implementations/hdk_on_native/test/utils.py:57: in hdk_comparator
    with ForceHdkImport(df1, df2):
modin/experimental/core/execution/native/implementations/hdk_on_native/test/utils.py:169: in __init__
    frame_id = DbWorker().import_pandas_dataframe(frame)
modin/experimental/core/execution/native/implementations/hdk_on_native/base_worker.py:254: in import_pandas_dataframe
    return cls.import_arrow_table(pa.Table.from_pandas(df))
pyarrow/table.pxi:3557: in pyarrow.lib.Table.from_pandas
    ???
../../micromamba/envs/modin_on_hdk/lib/python3.8/site-packages/pyarrow/pandas_compat.py:611: in dataframe_to_arrays
    arrays = [convert_column(c, f)
../../micromamba/envs/modin_on_hdk/lib/python3.8/site-packages/pyarrow/pandas_compat.py:611: in <listcomp>
    arrays = [convert_column(c, f)
../../micromamba/envs/modin_on_hdk/lib/python3.8/site-packages/pyarrow/pandas_compat.py:598: in convert_column
    raise e
../../micromamba/envs/modin_on_hdk/lib/python3.8/site-packages/pyarrow/pandas_compat.py:592: in convert_column
    result = pa.array(col, type=type_, from_pandas=True, safe=safe)
pyarrow/array.pxi:316: in pyarrow.lib.array
    ???
pyarrow/array.pxi:83: in pyarrow.lib._ndarray_to_array
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   pyarrow.lib.ArrowTypeError: ("Expected bytes, got a 'int' object", 'Conversion failed for column _I0 with type object')

pyarrow/error.pxi:123: ArrowTypeError


```

</details>


### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

--- a/modin/pandas/test/test_io.py
+++ b/modin/pandas/test/test_io.py
@@ -1892,7 +1892,7 @@ class TestExcel:
             io="modin/pandas/test/data/every_other_row_nan.xlsx",
         )
 
-    @pytest.mark.skipif(
+    @pytest.mark.xfail(
         StorageFormat.get() == "Hdk",
         reason="The frame contains different dtypes in the same column and could not be converted to arrow",
     )

--- a/modin/pandas/test/test_io.py
+++ b/modin/pandas/test/test_io.py
@@ -1892,6 +1892,10 @@ class TestExcel:
             io="modin/pandas/test/data/every_other_row_nan.xlsx",
         )
 
+    @pytest.mark.skipif(
+        StorageFormat.get() == "Hdk",
+        reason="The frame contains different dtypes in the same column and could not be converted to arrow",
+    )
     @check_file_leaks
     def test_read_excel_header_none(self):
         eval_io(

